### PR TITLE
close account w/tests

### LINKED ISSUE #23930: Close Organization Account on Deletion
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Accounts can now be deleted as a matter of closing accounts. https://github.com/aws/aws-sdk-go/pull/4338

Currently accounts are just [removed from the org](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/organizations/account.go#L279-L281). With the new API they can be deleted.

### New or Affected Resource(s)

* aws_organizations_account

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2022/03/aws-organizations-central-account-closure-lifecycle-management/).
[Blog post](https://aws.amazon.com/blogs/mt/aws-organizations-now-provides-a-simple-scalable-and-more-secure-way-to-close-your-member-accounts/).

### PR DESCRIPTION
**Pull request by danquack**
This PR is a continuation branch of #23930 with the recommended changes for testing. Since I have exceeded my quota on my personal accounts, I can not run this. This also increases the tshirt size so feel free to consider that when reviewing.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #23930
Relates OR Closes #23931

Output from acceptance testing:

```
$ export TEST_AWS_ORGANIZATION_ACCOUNT_EMAIL_DOMAIN="amazon.com"
$ make testacc TESTARGS="-run=TestAccAccount_" PKG_NAME="./internal/service/organizations" ACCTEST_PARALLELISM=1
## Can not execute due to quota exceeded
...
```

### CODE REVIEW

File: website/docs/r/organizations_account.html.markdown [hunk 1]
```diff
@@ -32,6 +33,7 @@ The following arguments are supported:
...
 * `parent_id` - (Optional) Parent Organizational Unit ID or Root ID for the account. Defaults to the Organization default Root ID. A configuration must be present for this argument to perform drift detection.
 * `role_name` - (Optional) The name of an IAM role that Organizations automatically preconfigures in the new member account. This role trusts the master account, allowing users in the master account to assume the role, as permitted by the master account administrator. The role has administrator permissions in the new member account. The Organizations API provides no method for reading this information after account creation, so Terraform cannot perform drift detection on its value and will always show a difference for a configured value after import unless [`ignore_changes`](https://www.terraform.io/docs/configuration/meta-arguments/lifecycle.html#ignore_changes) is used.
+* `close_account_on_deletion` - (Optional) If true, a deletion event will close the account. Otherwise, it will only remove from the organization.
```
Comment (drewmullen) [inline website/docs/r/organizations_account.html.markdown:36]: should be `close_on_deletion` i believe

Comment (danquack) [reply website/docs/r/organizations_account.html.markdown:36]: whoops. Ill fix that in both!

File: internal/service/organizations/account_test.go [hunk 2]
```diff
@@ -14,8 +14,7 @@ import (
 	"github.com/hashicorp/terraform-provider-aws/internal/conns"
 )
 
-func testAccAccount_basic(t *testing.T) {
-	acctest.Skip(t, "AWS Organizations Account testing is not currently automated due to manual account deletion steps.")
```
Comment (drewmullen) [inline internal/service/organizations/account_test.go:18]: Can you please add environment varlables instead of the Skips

**Comment by danquack** [comment]
Im just going to merge this into #23930 and can work off of that.

**Pull request closed without merging**